### PR TITLE
code cleanup , added function to return HTTP status code based on error

### PR DESCRIPTION
**Pull request by Madhu-1**
Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@prashanthpai please review

File: glusterd2/servers/rest/utils/utils.go [hunk 1]
```diff
@@ -75,3 +77,20 @@ func SendHTTPError(ctx context.Context, w http.ResponseWriter, statusCode int,
 		logger.WithError(err).Error("Failed to send the response -", resp)
 	}
 }
+
+// GetStatusCode returns error and http Status code based on  err
```
Comment (prashanthpai) [inline glusterd2/servers/rest/utils/utils.go]: A better name would be `StatusCodeFromErr` or  `ErrToStatusCode`

File: glusterd2/commands/snapshot/snapshot-create.go [hunk 2]
```diff
@@ -656,22 +656,16 @@ func snapshotCreateHandler(w http.ResponseWriter, r *http.Request) {
...
 	defer txn.Done()
 
 	vol, e := volume.GetVolume(req.VolName)
 	if e != nil {
-		if e == gderrors.ErrVolNotFound {
-			restutils.SendHTTPError(ctx, w, http.StatusNotFound, gderrors.ErrVolNotFound)
-		} else {
-			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, e)
-		}
+		status, err := restutils.GetStatusCode(err)
```
Comment (prashanthpai) [inline glusterd2/commands/snapshot/snapshot-create.go]: Incorrect variable is used here. The error variable here is named `e`. You have to change that to `err` and pass that

File: plugins/georeplication/rest.go [hunk 3]
```diff
@@ -112,23 +112,17 @@ func georepCreateHandler(w http.ResponseWriter, r *http.Request) {
...
 
 	// Check if Master volume exists and Matches with passed Volume ID
 	vol, e := volume.GetVolume(req.MasterVol)
 	if e != nil {
-		if e == errors.ErrVolNotFound {
-			restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound)
-		} else {
-			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, e)
-		}
+		status, e := restutils.GetStatusCode(e)
```
Comment (prashanthpai) [inline plugins/georeplication/rest.go]: Suggestion unrelated to your PR:

For code uniformity, you can name this variable as `err`.

Please do the same for other occurrences of `e` in the file.

**Comment by Madhu-1** [comment]
retest this please

**Comment by Madhu-1** [comment]
retest this please

**Comment by prashanthpai** [comment]
@Madhu-1 

This seems like a genuine failure:

```log
06:46:23 ERROR: failing tests:
06:46:23   ./test/020-golint.sh
06:46:23 make: *** [test] Error 1
```

Please ensure that you run `make test` locally before sending PR.

**Comment by prashanthpai** [comment]
@Madhu-1 There's no need for this PR to contain two separate commits - one for adding the function and another for using it.

**Comment by Madhu-1** [comment]
@prashanthpai  squashed multiple commits into one commit.

**Comment by prashanthpai** [comment]
retest this please

**Pull request merged**